// deezer-node/src/index.js
var rp = require("request-promise");
var ErrorOcurred = require("./error");

let url;

function Deezer() {
  this.apiUrl = "https://api.deezer.com/";
}

Deezer.prototype.getAlbum = function(albumId) {
  url = "album/" + albumId;
  console.log(this);
  return rp({ uri: this.apiUrl + url, json: true });
};

Deezer.prototype.getArtist = function(artistId) {
  url = "artist/" + artistId;
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.getChart = function() {
  url = "chart";
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.getAlbumComments = function(albumId) {
  url = "album/" + albumId + "/comments";
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.getArtistComments = function(artistId) {
  url = "album/" + artistId + "/comments";
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.getPlaylistComments = function(playlistId) {
  url = "playlist/" + playlistId + "/comments";
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.getEditorial = function(mode) {
  if (mode === "selection") {
    url = "editorial/0/selection";
  } else if (mode === "charts") {
    url = "editorial/0/charts";
  } else if (mode === "releases") {
    url = "editorial/0/releases";
  } else {
    throw new ErrorOcurred(
      "You need to use any of these values: ​​" +
        "selection, " +
        "charts, " +
        "releases"
    );
  }
  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.findTracks = function(options, index, order) {
  var url = "search?q=";
  var query = "";
  if (typeof options === "object") {
    for (var key in options) {
      query = query + key + ':"' + options[key] + '" ';
    }
  } else {
    query = options;
  }
  url = url + query;

  if (index !== 0) url = url + "&index=" + index;
  if (order) url = url + "&order=" + order;

  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.findAlbums = function(query, index) {
  var url = "search/album?q=" + query;
  if (index !== 0) url = url + "&index=" + index;

  return rp({ url: this.apiUrl + url, json: true });
};

Deezer.prototype.findArtists = function(query, index) {
  var url = "search/artist?q=" + query;
  if (index !== 0) url = url + "&index=" + index;

  return rp({ url: this.apiUrl + url, json: true });
};

module.exports = Deezer;
